lib/common/api/deprecate.js

```diff
@@ -46,6 +46,26 @@ deprecate.log = (message) => {
   }
 }
 
+// Deprecate an event.
+deprecate.event = (emitter, oldName, newName, fn) => {
+  let warned = false
+  return emitter.on(newName, function (...args) {
+    // There are no listeners for this event
+    if (this.listenerCount(oldName) === 0) { return }
+    // noDeprecation set or if user has already been warned
+    if (warned || process.noDeprecation) { return }
+    warned = true
+    // Event name starts with a -, which means it's an internal event.
+    // For deprecation, that means the event cannot be used anymore.
+    if (!newName.startsWith('-')) { deprecate.warn(`'${oldName}' event`, `'${newName}' event`) } else { deprecate.log(`'${oldName}' event has been deprecated.`) }
+    if (fn != null) {
+      fn.apply(this, arguments)
+    } else {
+      this.emit.apply(this, [oldName].concat(args))
```

Comment:
`this.emit.call(this, oldName, ...args)`

Reply:
Why is call better than apply?

Reply:
@nitsakh 
You probably somehow saw the very first version of my suggestion )
`this.emit(oldName, ...args)` is much better alternative here.

----

_But if you're curious_
> Why is `call` better than `apply`?

In general there's no difference.
But `this.emit.call(this, oldName, ...args)` is better than `this.emit.apply(this, [oldName].concat(args))` because the second option makes you _think_ more and that's not what you should be doing while reading code =)